source/extensions/filters/network/dubbo_proxy/filter.cc

```diff
@@ -0,0 +1,212 @@
+    return;
+  }
+
+  if (event == Network::ConnectionEvent::RemoteClose) {
+    stats_.cx_destroy_local_with_active_rq_.inc();
+  }
+
+  if (event == Network::ConnectionEvent::LocalClose) {
+    stats_.cx_destroy_remote_with_active_rq_.inc();
+  }
+}
+
+void Filter::onRequestMessage(RequestMessagePtr&& message) {
+  ASSERT(message);
+  ASSERT(message->messageType() == MessageType::Request);
+
+  stats_.request_.inc();
+  message->isTwoWay() ? stats_.request_twoway_.inc() : stats_.request_oneway_.inc();
+
+  if (message->isEvent()) {
+    stats_.request_event_.inc();
+  }
+
+  ENVOY_LOG(debug, "dubbo request: started {} message", message->requestId());
+
+  // One-way messages do not receive responses.
+  if (!message->isTwoWay()) {
+    return;
+  }
+
+  auto request = std::make_unique<ActiveMessage>(*this, message->requestId());
+  active_call_map_.emplace(message->requestId(), std::move(request));
+}
+
+void Filter::onResponseMessage(ResponseMessagePtr&& message) {
+  ASSERT(message);
+  ASSERT(message->messageType() == MessageType::Response);
+
+  auto itor = active_call_map_.find(message->requestId());
+  if (itor == active_call_map_.end()) {
+    throw EnvoyException(fmt::format("unknown request id {}", message->requestId()));
+  }
+  active_call_map_.erase(itor);
+
+  ENVOY_LOG(debug, "dubbo response: ended {} message", message->requestId());
+
+  stats_.response_.inc();
+  switch (message->responseStatus()) {
+  case ResponseStatus::Ok:
+    stats_.response_success_.inc();
+    break;
+  default:
+    stats_.response_error_.inc();
+    ENVOY_LOG(error, "dubbo response status: {}", static_cast<uint8_t>(message->responseStatus()));
+    break;
+  }
+}
+
+void Filter::onRpcInvocation(RpcInvocationPtr&& invo) {
+  ENVOY_LOG(debug, "dubbo request: method name is {}, service name is {}, service version {}",
+            invo->getMethodName(), invo->getServiceName(), invo->getServiceVersion());
+}
+
+void Filter::onRpcResult(RpcResultPtr&& res) {
+  if (res->hasException()) {
+    stats_.response_exception_.inc();
+  }
+}
+
+DubboFilterStats Filter::generateStats(const std::string& prefix, Stats::Scope& scope) {
+  return DubboFilterStats{ALL_DUBBO_FILTER_STATS(POOL_COUNTER_PREFIX(scope, prefix),
+                                                 POOL_GAUGE_PREFIX(scope, prefix),
+                                                 POOL_HISTOGRAM_PREFIX(scope, prefix))};
+}
+
+DecoderPtr Filter::createDecoder(ProtocolCallbacks& prot_callback) {
+  auto parser = createProtocol(prot_callback);
+  auto serializer = createDeserializer();
+  return std::make_unique<Decoder>(std::move(parser), std::move(serializer), *this);
+}
+
+ProtocolPtr Filter::createProtocol(ProtocolCallbacks& callback) {
+  using Type = envoy::extensions::filters::network::dubbo_proxy::v2alpha1::DubboProxy;
+  switch (protocol_type_) {
+  case Type::Dubbo:
+    return std::make_unique<DubboProtocolImpl>(callback);
+  default:
+    throw EnvoyException(
```

Comment:
missing coverage, if this cannot happen, just use `NOT_REACHED_GCOVR_EXCL_LINE`, same for below

Reply:
fixed